### ISSUE TITLE
Slider not responding to diagonal dragging

### ISSUE THREAD

**Issue by morten-hogseth** [akiran/react-slick#1031]
### Hi, my issue is regarding diagonal drag events.

Looking at your example site http://neostack.com/opensource/react-slick. And then at the first example "Single Item", we see that this slider responds very well to drag events, diagonal as well.

But when I am using the library in my react project I notice it does not respond as well to drag events.
Then I went to test out your code sandbox example. And I can see that the issue remains there as well.

So, I have found old examples where the slider responds as I expect, ref: https://codepen.io/spencercarnage/pen/VjNWPj. So then I wonder if this is related to any updates done recently?

Here is the code sandbox where the issue can be seen when trying to drag diagonally.
https://codesandbox.io/s/zzloxr09mp

I have only tested this in Chrome.

**Comment by Sjiep** [comment]
I was searching this exact same problem and found this issue in the process.

In the examples at http://neostack.com/opensource/react-slick the swiping of a slide is still triggered if you drag the mouse or finger first in a vertical direction and then horizontally, whereas this will fail in the code sandbox example. I.e. the horizontal movement of the dragging motion will no longer trigger the swiping of the slide when it has detected a vertical motion first.

The result of this behaviour is that (for me) in like 20% of the cases the swiping of a slide fails giving an unpleasant user experience.

**Comment by laveesingh** [comment]
That is intended. If you try vertical dragging for more than 4px, it'll determine it as a vertical event. Until this event ends, horizontal scrolling cannot occur. Although this resistance limit of vertical dragging has been increased from 4px to 10px. b9481f2ce8676725a41b6dfe9847cb8c761eb277

**Comment by Sjiep** [comment]
I'm guess this feature was implemented to prevent scrolling of the page and swiping of a slide at the same time?

It doesn't look like there is a way to change this value through property. As I said, with the current setting the horizontal scroll event is quite frequently not triggered, for something what feels like a 'legit' horizontal swipe. Often even occurring multiple times in a row.

On other note, I see the change from 4px to 10px is a commit from only an hour ago. I just tested it by locally changing it from 4 to 10 and the result is actually a lot better!
